stock_provider_ref_on_receipt/models/stock.py

```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+from openerp import api, fields, models
+
+
+class Product(models.Model):
+    _inherit = 'stock.pack.operation'
+
+    provider_ref = fields.Char(string="Provider Reference",
+                               compute="_compute_product_code")
+
+    @api.multi
+    @api.depends('provider_ref')
+    def _compute_product_code(self):
```

Comment:
Here, I think you should write : 
```python
for line in self:
    product_supplier = self.env['product.supplierinfo'].search(
            [('product_tmpl_id', '=', self.product_id.product_tmpl_id.id)])
    line.provider_ref = product_supplier[0].product_code
```
Because `self.product_id` will not work if there is more than one record in recordset.

Reply:
Done

Reply:
just note that if there is no supplier set on the product doing product_supplier[0] may rise an exception error.

Reply:
shouldn't it be **line**.product_id.product_tmpl_id.id ?

```python
for line in self:
    product_supplier = self.env['product.supplierinfo'].search(
            [('product_tmpl_id', '=', line.product_id.product_tmpl_id.id)])
    line.provider_ref = product_supplier[0].product_code
```

Reply:
@houssine78  is it correct like this ?
@robinkeunen  Could you precise ? isn't already line.product_id.product_tmpl_id.id ?

Reply:
Maybe you forgot to push but Rémy's version and your version are `self.product_id.product_tmpl_id.id)])`

+ Houssine comments

Reply:
nice catch @robinkeunen

Reply:
Done